src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/heatmap.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { omit } from 'lodash';
+
+import { schema, type TypeOf } from '@kbn/config-schema';
+
+import { datasetSchema, datasetEsqlTableSchema } from '../dataset';
+import { colorByValueSchema } from '../color';
+import { esqlColumnSchema, metricOperationDefinitionSchema } from '../metric_ops';
+import { sharedPanelInfoSchema, layerSettingsSchema } from '../shared';
+import { mergeAllMetricsWithChartDimensionSchema } from './shared';
+import { positionSchema } from '../alignments';
+
+export const titleSchemaProps = {
+  value: schema.maybe(schema.string({ defaultValue: '' })),
+  visible: schema.boolean({ defaultValue: false }),
+};
+
+const legendSchemaProps = {
+  truncate_after_lines: schema.maybe(schema.number()),
+  visible: schema.maybe(schema.boolean()),
+  size: schema.maybe(
+    schema.oneOf(
+      [
+        schema.literal('auto'),
+        schema.literal('small'),
+        schema.literal('medium'),
+        schema.literal('large'),
+        schema.literal('xlarge'),
+      ],
+      { defaultValue: 'auto' }
+    )
+  ),
```

Comment:
Could we align this with...

https://github.com/elastic/kibana/blob/c152f868f796664103230409cfaba325ae4c7b59/src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/legacy_metric.ts#L29-L41